tools/mkclass/classcompiler.cpp

```diff
@@ -379,13 +379,19 @@ void ClassCompiler::HandleClass(const Klass& klass, const ClassDebugInfo&)
 	m_Header << "\t" << "const std::unordered_set<Type*>& GetLoadDependencies() const override;" << std::endl;
 
 	m_Impl << "const std::unordered_set<Type*>& TypeImpl<" << klass.Name << ">::GetLoadDependencies() const" << std::endl
-		<< "{" << std::endl
-		<< "\t" << "static const std::unordered_set<Type*> deps ({" << std::endl;
+		<< "{" << std::endl;
+
+	for (auto& dep : klass.LoadDependencies)
+		m_Impl << "\tstatic const auto type" << dep << " (GetByName(\"" << dep << "\").get());" << std::endl
+			<< "\tstatic const auto v" << dep << " (VERIFY(type" << dep << "));" << std::endl
+			<< "\t(void)v" << dep << ";" << std::endl << std::endl;
```

Comment:
This is not what I intended with my previous comment, but an actual for loop that is being executed when `GetLoadDependencies()` is called. Can you please change this to produce an actual implementation like this: e.g. `Service`:
```cxx
static std::unordered_set<Type*> deps;
// Render this if block only when the type actually has a load_after dependencies, i.e. if not klass.LoadDependencies.empty()
if(deps.empty()) { // Only if deps is not yet initialised
	deps.reserve(4); // Reserve the size of the dependencies "klass.LoadDependencies.size()"

	for (auto& name : {"ApiListener","Endpoint","Host","Zone",}) {
		auto dep (GetByName(name).get());
		VERIFY_MSG(dep != nullptr, "type 'Service' load depends on non-existent type 'type" + name + "'.");
		deps.emplace(dep);
	}
}

return deps;
```

Reply:
But right now VERIFY() perfectly indicates which type name doesn't exist – via the expression which is the var name. In a for loop, however... 🤷‍♂️

Reply:
If you do not want to introduce the `VERIFY_MSG`, then please at least make sure that you do not always call `VERIFY`, as all variables are static, that the sanity check should only take place once as well.
```cxx
static std::unordered_set<Type*> deps;
if (deps.empty) {
    auto typeListener ....;
    VERIFY(typeListener != nullptr);
     ...

   deps.emplace(typeListener, typeThis, typeOther, ...)
}

return deps;
```

Reply:
That's already the case, as the VERIFY() expression is inside a static var constructor.

Reply:
This is exactly the part that I don't understand why you're doing so, instead of wrapping them in a simple if statement.

Reply:
Caching each `VERIFY` result of dep type just clutters up the code even more.